tools/plugins/com.liferay.ide.server.tomcat.core/src/com/liferay/ide/server/tomcat/core/PortalTomcatBundle.java

```diff
@@ -77,13 +79,35 @@ public IPath getAppServerLibGlobalDir() {
 
 	@Override
 	public IPath getAppServerPortalDir() {
-		IPath retval = null;
+		IPath appBasePath = bundlePath.append("webapps");
```

Comment:
This method now performs a calculation for each time it is called.  Instead it should inintialize a appServerPortalDIr the first time it is accessed but then should be cached there-after.  If the bundlePath changes, then it should re-calculate.